{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ./0.0-gks-utility-functions.ipynb\n",
    "%run ./0.1-gks-matplotlib-plots.ipynb\n",
    "%run 0.2-gks-setup-jupyter-notebook-view.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "files = os.listdir(DATA_INTERIM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_data_files = [x for x in files if x[0:10]==\"meta_data_\" and x.endswith(\".csv\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kwargs:  {}\n",
      "DataFrame Size: (176, 9)\n",
      "DataFrame Columns: Index(['cord_uid', 'source_x', 'title', 'abstract', 'publish_time', 'authors',\n",
      "       'has_pdf_parse', 'has_pmc_xml_parse', 'full_text_file'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>has_pdf_parse</th>\n",
       "      <th>has_pmc_xml_parse</th>\n",
       "      <th>full_text_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>laawgrxl</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Transmission Pathways of Foot-and-Mouth Diseas...</td>\n",
       "      <td>Foot-and-mouth disease (FMD) virus causes an a...</td>\n",
       "      <td>2008-04-18</td>\n",
       "      <td>Cottam, Eleanor M.; Wadsworth, Jemma; Shaw, An...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5ns2t5i2</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Safety and immunogenicity of an MF59™-adjuvant...</td>\n",
       "      <td>BACKGROUND: The safety and immunogenicity of a...</td>\n",
       "      <td>2008-02-20</td>\n",
       "      <td>Li, Rongcheng; Fang, Hanhua; Li, Yanping; Liu,...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4tmqx6wr</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Antibodies to Nipah or Nipah-like Viruses in B...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-12-23</td>\n",
       "      <td>Li, Yan; Wang, Jianmin; Hickey, Andrew C.; Zha...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ursd3h22</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Human Influenza A (H5N1) Cases, Urban Areas of...</td>\n",
       "      <td>We investigated potential sources of infection...</td>\n",
       "      <td>2007-07-23</td>\n",
       "      <td>Yu, Hongjie; Feng, Zijian; Zhang, Xianfeng; Xi...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9vpzk6wu</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Profiling of cellular proteins in porcine repr...</td>\n",
       "      <td>BACKGROUND: Porcine reproductive and respirato...</td>\n",
       "      <td>2010-09-18</td>\n",
       "      <td>Zhang, Chengwen; Xue, Chunyi; Li, Yan; Kong, Q...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>76ia62hl</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Characterization of protein–protein interactio...</td>\n",
       "      <td>Abstract The human coronavirus, associated wit...</td>\n",
       "      <td>2004-10-31</td>\n",
       "      <td>He, Runtao; Leeson, Andrew; Ballantine, Meliss...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>jv5wyup8</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Production of a monoclonal antibody against SA...</td>\n",
       "      <td>Abstract Severe acute respiratory syndrome (SA...</td>\n",
       "      <td>2005-09-15</td>\n",
       "      <td>Yu, Xiao Fei; Liang, Lin Hui; She, Ming; Liao,...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>dtwuav65</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>A novel pyrosequencing assay for the detection...</td>\n",
       "      <td>Abstract A novel reassortant avian influenza A...</td>\n",
       "      <td>2014-01-22</td>\n",
       "      <td>Qi, Yuhua; Fan, Huan; Qi, Xian; Zhu, Zheng; Gu...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>i738t9lf</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Genome Organization of the SARS-CoV</td>\n",
       "      <td>Annotation of the genome sequence of the SARS-...</td>\n",
       "      <td>2016-11-28</td>\n",
       "      <td>Xu, Jing; Hu, Jianfei; Wang, Jing; Han, Yujun;...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>noncomm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>rnaktik9</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Structural model of the SARS coronavirus E cha...</td>\n",
       "      <td>Abstract Coronaviruses (CoV) cause common cold...</td>\n",
       "      <td>2018-06-30</td>\n",
       "      <td>Surya, Wahyu; Li, Yan; Torres, Jaume</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>176 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cord_uid  source_x                                              title  \\\n",
       "0    laawgrxl       PMC  Transmission Pathways of Foot-and-Mouth Diseas...   \n",
       "1    5ns2t5i2       PMC  Safety and immunogenicity of an MF59™-adjuvant...   \n",
       "2    4tmqx6wr       PMC  Antibodies to Nipah or Nipah-like Viruses in B...   \n",
       "3    ursd3h22       PMC  Human Influenza A (H5N1) Cases, Urban Areas of...   \n",
       "4    9vpzk6wu       PMC  Profiling of cellular proteins in porcine repr...   \n",
       "..        ...       ...                                                ...   \n",
       "171  76ia62hl  Elsevier  Characterization of protein–protein interactio...   \n",
       "172  jv5wyup8  Elsevier  Production of a monoclonal antibody against SA...   \n",
       "173  dtwuav65  Elsevier  A novel pyrosequencing assay for the detection...   \n",
       "174  i738t9lf       PMC                Genome Organization of the SARS-CoV   \n",
       "175  rnaktik9  Elsevier  Structural model of the SARS coronavirus E cha...   \n",
       "\n",
       "                                              abstract publish_time  \\\n",
       "0    Foot-and-mouth disease (FMD) virus causes an a...   2008-04-18   \n",
       "1    BACKGROUND: The safety and immunogenicity of a...   2008-02-20   \n",
       "2                                                  NaN   2008-12-23   \n",
       "3    We investigated potential sources of infection...   2007-07-23   \n",
       "4    BACKGROUND: Porcine reproductive and respirato...   2010-09-18   \n",
       "..                                                 ...          ...   \n",
       "171  Abstract The human coronavirus, associated wit...   2004-10-31   \n",
       "172  Abstract Severe acute respiratory syndrome (SA...   2005-09-15   \n",
       "173  Abstract A novel reassortant avian influenza A...   2014-01-22   \n",
       "174  Annotation of the genome sequence of the SARS-...   2016-11-28   \n",
       "175  Abstract Coronaviruses (CoV) cause common cold...   2018-06-30   \n",
       "\n",
       "                                               authors  has_pdf_parse  \\\n",
       "0    Cottam, Eleanor M.; Wadsworth, Jemma; Shaw, An...           True   \n",
       "1    Li, Rongcheng; Fang, Hanhua; Li, Yanping; Liu,...           True   \n",
       "2    Li, Yan; Wang, Jianmin; Hickey, Andrew C.; Zha...           True   \n",
       "3    Yu, Hongjie; Feng, Zijian; Zhang, Xianfeng; Xi...           True   \n",
       "4    Zhang, Chengwen; Xue, Chunyi; Li, Yan; Kong, Q...           True   \n",
       "..                                                 ...            ...   \n",
       "171  He, Runtao; Leeson, Andrew; Ballantine, Meliss...           True   \n",
       "172  Yu, Xiao Fei; Liang, Lin Hui; She, Ming; Liao,...           True   \n",
       "173  Qi, Yuhua; Fan, Huan; Qi, Xian; Zhu, Zheng; Gu...           True   \n",
       "174  Xu, Jing; Hu, Jianfei; Wang, Jing; Han, Yujun;...           True   \n",
       "175               Surya, Wahyu; Li, Yan; Torres, Jaume           True   \n",
       "\n",
       "     has_pmc_xml_parse      full_text_file  \n",
       "0                 True     comm_use_subset  \n",
       "1                 True     comm_use_subset  \n",
       "2                 True      custom_license  \n",
       "3                 True      custom_license  \n",
       "4                 True     comm_use_subset  \n",
       "..                 ...                 ...  \n",
       "171              False      custom_license  \n",
       "172              False      custom_license  \n",
       "173              False      custom_license  \n",
       "174               True  noncomm_use_subset  \n",
       "175              False      custom_license  \n",
       "\n",
       "[176 rows x 9 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "detail_csv_file(DATA_INTERIM + meta_data_files[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NLP Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kwargs:  {}\n",
      "DataFrame Size: (176, 9)\n",
      "DataFrame Columns: Index(['cord_uid', 'source_x', 'title', 'abstract', 'publish_time', 'authors',\n",
      "       'has_pdf_parse', 'has_pmc_xml_parse', 'full_text_file'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "d = detail_csv_file(DATA_INTERIM + meta_data_files[0])\n",
    "corpus = d.fillna(\"\").abstract.tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "176"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(corpus)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Common ngrams analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [],
   "source": [
    "vocab=[\"aerosol\", \"droplet\", \"apex\", \"asymptomatic\", \"chloroquine\", \"community spread\", \"communicable\",\n",
    "      \"coronavirus\", \"exponential\", \"immunity\", \"immunocompromised\", \"incubation\", \"mitigation\", \"covid-19\",\n",
    "      \"covid\", \"covid 19\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "pattern = \"(?u)\\\\b[\\'\\w-]+\\\\b\"\n",
    "# vectorizer = CountVectorizer(stop_words='english', token_pattern=pattern, min_df=0.1)\n",
    "vectorizer = CountVectorizer(stop_words='english', token_pattern=pattern, vocabulary=vocab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['aerosol', 'droplet', 'apex', 'asymptomatic', 'chloroquine', 'community spread', 'communicable', 'coronavirus', 'exponential', 'immunity', 'immunocompromised', 'incubation', 'mitigation', 'covid-19', 'covid', 'covid 19']\n"
     ]
    }
   ],
   "source": [
    "X = vectorizer.fit_transform(corpus)\n",
    "print(vectorizer.get_feature_names())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# vectorizer.stop_words_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(176, 16)"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(176, 16)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>aerosol</th>\n",
       "      <th>droplet</th>\n",
       "      <th>apex</th>\n",
       "      <th>asymptomatic</th>\n",
       "      <th>chloroquine</th>\n",
       "      <th>community spread</th>\n",
       "      <th>communicable</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>exponential</th>\n",
       "      <th>immunity</th>\n",
       "      <th>immunocompromised</th>\n",
       "      <th>incubation</th>\n",
       "      <th>mitigation</th>\n",
       "      <th>covid-19</th>\n",
       "      <th>covid</th>\n",
       "      <th>covid 19</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   aerosol  droplet  apex  asymptomatic  chloroquine  community spread  \\\n",
       "0  0        0        0     0             0            0                  \n",
       "1  0        0        0     0             0            0                  \n",
       "2  0        0        0     0             0            0                  \n",
       "3  0        0        0     0             0            0                  \n",
       "4  0        0        0     0             0            0                  \n",
       "\n",
       "   communicable  coronavirus  exponential  immunity  immunocompromised  \\\n",
       "0  0             0            0            0         0                   \n",
       "1  0             0            0            0         0                   \n",
       "2  0             0            0            0         0                   \n",
       "3  0             0            0            0         0                   \n",
       "4  0             0            0            0         0                   \n",
       "\n",
       "   incubation  mitigation  covid-19  covid  covid 19  \n",
       "0  0           0           0         0      0         \n",
       "1  0           0           0         0      0         \n",
       "2  0           0           0         0      0         \n",
       "3  0           0           0         0      0         \n",
       "4  0           0           0         0      0         "
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "feature_index_map = vectorizer.vocabulary_\n",
    "cols = sorted(feature_index_map, key=feature_index_map.get)\n",
    "df = pd.DataFrame(X.toarray(), columns=cols)\n",
    "\n",
    "print(df.shape)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['aerosol', 'droplet', 'apex', 'asymptomatic', 'chloroquine',\n",
       "       'community spread', 'communicable', 'coronavirus', 'exponential',\n",
       "       'immunity', 'immunocompromised', 'incubation', 'mitigation', 'covid-19',\n",
       "       'covid', 'covid 19'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'asymptomatic' mentioned '2' times\n",
      "'coronavirus' mentioned '58' times\n",
      "'immunity' mentioned '11' times\n",
      "'incubation' mentioned '1' times\n",
      "'covid-19' mentioned '15' times\n",
      "'covid' mentioned '3' times\n"
     ]
    }
   ],
   "source": [
    "useful_cols = list()\n",
    "\n",
    "for col in df.columns:\n",
    "    if df[col].sum()>0:\n",
    "        print(\"'{}' mentioned '{}' times\".format(col, df[col].sum()))\n",
    "        useful_cols.append(col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(52, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>asymptomatic</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>immunity</th>\n",
       "      <th>incubation</th>\n",
       "      <th>covid-19</th>\n",
       "      <th>covid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    asymptomatic  coronavirus  immunity  incubation  covid-19  covid\n",
       "8   1             0            0         0           0         0    \n",
       "18  0             1            1         0           0         0    \n",
       "20  0             1            0         0           0         0    \n",
       "21  0             1            0         0           0         0    \n",
       "22  0             1            1         0           0         0    "
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_term_search = df[useful_cols][df.apply(sum, axis=1).apply(bool)]\n",
    "\n",
    "print(df_term_search.shape)\n",
    "df_term_search.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "52"
      ]
     },
     "execution_count": 276,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "number_papers_with_vocab_present = df[useful_cols].apply(sum, axis=1).apply(bool).sum()\n",
    "number_papers_with_vocab_present"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(52, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>has_pdf_parse</th>\n",
       "      <th>has_pmc_xml_parse</th>\n",
       "      <th>full_text_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>gl9cjmno</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China</td>\n",
       "      <td>We estimated the attack rate of pandemic (H1N1) 2009 and assessed risk factors for infection among close contacts quarantined in Beijing, People’s Republic of China. The first 613 confirmed cases detected between May 16 and September 15, 2009, were investigated; 7,099 close contacts were located and quarantined. The attack rate of confirmed infection in close contacts was 2.4% overall, ranging from 0.9% among aircraft passengers to &gt;5% among household members. Risk factors for infection among close contacts were younger age, being a household member of an index case-patient, exposure during the index case-patient’s symptomatic phase, and longer exposure. Among close contacts with positive test results at the start of quarantine, 17.2% had subclinical infection. Having contact with a household member and younger age were the major risk factors for acquiring pandemic (H1N1) 2009 influenza virus infection. One person in 6 with confirmed pandemic (H1N1) 2009 was asymptomatic.</td>\n",
       "      <td>2011-10-23</td>\n",
       "      <td>Pang, Xinghuo; Yang, Peng; Li, Shuang; Zhang, Li; Tian, Lili; Li, Yang; Liu, Bo; Zhang, Yi; Liu, Baiwei; Huang, Ruogang; Li, Xinyu; Wang, Quanyi</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>p350i54m</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The role of C5a in acute lung injury induced by highly pathogenic viral infections</td>\n",
       "      <td>The complement system, an important part of innate immunity, plays a critical role in pathogen clearance. Unregulated complement activation is likely to play a crucial role in the pathogenesis of acute lung injury (ALI) induced by highly pathogenic virus including influenza A viruses H5N1, H7N9, and severe acute respiratory syndrome (SARS) coronavirus. In highly pathogenic virus-induced acute lung diseases, high levels of chemotactic and anaphylatoxic C5a were produced as a result of excessive complement activaiton. Overproduced C5a displays powerful biological activities in activation of phagocytic cells, generation of oxidants, and inflammatory sequelae named “cytokine storm”, and so on. Blockade of C5a signaling have been implicated in the treatment of ALI induced by highly pathogenic virus. Herein, we review the literature that links C5a and ALI, and review our understanding of the mechanisms by which C5a affects ALI during highly pathogenic viral infection. In particular, we discuss the potential of the blockade of C5a signaling to treat ALI induced by highly pathogenic viruses.</td>\n",
       "      <td>2015-05-06</td>\n",
       "      <td>Wang, Renxi; Xiao, He; Guo, Renfeng; Li, Yan; Shen, Beifen</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>noncomm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>86t9chxe</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China</td>\n",
       "      <td>On 26 May 2015, an imported Middle East respiratory syndrome coronavirus (MERS-CoV) was identified in Guangdong Province, China, and found to be closely related to the MERS-CoV strain prevalent in South Korea. The full genome of the ChinaGD01 strain was sequenced and analyzed to investigate the epidemiology and evolution of MERS-CoV circulating in South Korea and China.</td>\n",
       "      <td>2015-08-13</td>\n",
       "      <td>Lu, Roujian; Wang, Yanqun; Wang, Wenling; Nie, Kai; Zhao, Yanjie; Su, Juan; Deng, Yao; Zhou, Weimin; Li, Yang; Wang, Huijuan; Wang, Wen; Ke, Changwen; Ma, Xuejun; Wu, Guizhen; Tan, Wenjie</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>kzblr469</td>\n",
       "      <td>PMC</td>\n",
       "      <td>A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein</td>\n",
       "      <td>The newly-emerging Middle East respiratory syndrome coronavirus (MERS-CoV) can cause severe and fatal acute respiratory disease in humans. Despite global efforts, the potential for an associated pandemic in the future cannot be excluded. The development of effective counter-measures is urgent. MERS-CoV-specific anti-viral drugs or vaccines are not yet available. Using the spike receptor-binding domain of MERS-CoV (MERS-RBD) to immunize mice, we identified two neutralizing monoclonal antibodies (mAbs) 4C2 and 2E6. Both mAbs potently bind to MERS-RBD and block virus entry in vitro with high efficacy. We further investigated their mechanisms of neutralization by crystallizing the complex between the Fab fragments and the RBD, and solved the structure of the 4C2 Fab/MERS-RBD complex. The structure showed that 4C2 recognizes an epitope that partially overlaps the receptor-binding footprint in MERS-RBD, thereby interfering with the virus/receptor interactions by both steric hindrance and interface-residue competition. 2E6 also blocks receptor binding, and competes with 4C2 for binding to MERS-RBD. Based on the structure, we further humanized 4C2 by preserving only the paratope residues and substituting the remaining amino acids with the counterparts from human immunoglobulins. The humanized 4C2 (4C2h) antibody sustained similar neutralizing activity and biochemical characteristics to the parental mouse antibody. Finally, we showed that 4C2h can significantly abate the virus titers in lungs of Ad5-hCD26-transduced mice infected with MERS-CoV, therefore representing a promising agent for prophylaxis and therapy in clinical settings.</td>\n",
       "      <td>2015-09-22</td>\n",
       "      <td>Li, Yan; Wan, Yuhua; Liu, Peipei; Zhao, Jincun; Lu, Guangwen; Qi, Jianxun; Wang, Qihui; Lu, Xuancheng; Wu, Ying; Liu, Wenjun; Zhang, Buchang; Yuen, Kwok-Yung; Perlman, Stanley; Gao, George F; Yan, Jinghua</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>gmoq1i5x</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (MERS-CoV) with pandemic potential is a major worldwide threat to public health. However, vaccine development for this pathogen lags behind as immunity associated with protection is currently largely unknown. In this study, an immunoinformatics-driven genome-wide screening strategy of vaccine targets was performed to thoroughly screen the vital and effective dominant immunogens against MERS-CoV. Conservancy and population coverage analysis of the epitopes were done by the Immune Epitope Database. The results showed that the nucleocapsid (N) protein of MERS-CoV might be a better protective immunogen with high conservancy and potential eliciting both neutralizing antibodies and T-cell responses compared with spike (S) protein. Further, the B-cell, helper T-cell and cytotoxic T lymphocyte (CTL) epitopes were screened and mapped to the N protein. A total of 15 linear and 10 conformal B-cell epitopes that may induce protective neutralizing antibodies were obtained. Additionally, a total of 71 peptides with 9-mer core sequence were identified as helper T-cell epitopes, and 34 peptides were identified as CTL epitopes. Based on the maximum HLA binding alleles, top 10 helper T-cell epitopes and CTL epitopes that may elicit protective cellular immune responses against MERS-CoV were selected as MERS vaccine candidates. Population coverage analysis showed that the putative helper T-cell epitopes and CTL epitopes could cover the vast majority of the population in 15 geographic regions considered where vaccine would be employed. The B- and T-cell stimulation potentials of the screened epitopes is to be further validated for their efficient use as vaccines against MERS-CoV. Collectively, this study provides novel vaccine target candidates and may prompt further development of vaccines against MERS-CoV and other emerging infectious diseases.</td>\n",
       "      <td>2015-12-07</td>\n",
       "      <td>Shi, Jiandong; Zhang, Jing; Li, Sijin; Sun, Jing; Teng, Yumei; Wu, Meini; Li, Jianfan; Li, Yanhan; Hu, Ningzhu; Wang, Haixuan; Hu, Yunzhang</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    cord_uid source_x  \\\n",
       "8   gl9cjmno  PMC       \n",
       "18  p350i54m  PMC       \n",
       "20  86t9chxe  PMC       \n",
       "21  kzblr469  PMC       \n",
       "22  gmoq1i5x  PMC       \n",
       "\n",
       "                                                                                                                                       title  \\\n",
       "8   Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China                                                 \n",
       "18  The role of C5a in acute lung injury induced by highly pathogenic viral infections                                                         \n",
       "20  Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China         \n",
       "21  A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein                              \n",
       "22  Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          abstract  \\\n",
       "8   We estimated the attack rate of pandemic (H1N1) 2009 and assessed risk factors for infection among close contacts quarantined in Beijing, People’s Republic of China. The first 613 confirmed cases detected between May 16 and September 15, 2009, were investigated; 7,099 close contacts were located and quarantined. The attack rate of confirmed infection in close contacts was 2.4% overall, ranging from 0.9% among aircraft passengers to >5% among household members. Risk factors for infection among close contacts were younger age, being a household member of an index case-patient, exposure during the index case-patient’s symptomatic phase, and longer exposure. Among close contacts with positive test results at the start of quarantine, 17.2% had subclinical infection. Having contact with a household member and younger age were the major risk factors for acquiring pandemic (H1N1) 2009 influenza virus infection. One person in 6 with confirmed pandemic (H1N1) 2009 was asymptomatic.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "18  The complement system, an important part of innate immunity, plays a critical role in pathogen clearance. Unregulated complement activation is likely to play a crucial role in the pathogenesis of acute lung injury (ALI) induced by highly pathogenic virus including influenza A viruses H5N1, H7N9, and severe acute respiratory syndrome (SARS) coronavirus. In highly pathogenic virus-induced acute lung diseases, high levels of chemotactic and anaphylatoxic C5a were produced as a result of excessive complement activaiton. Overproduced C5a displays powerful biological activities in activation of phagocytic cells, generation of oxidants, and inflammatory sequelae named “cytokine storm”, and so on. Blockade of C5a signaling have been implicated in the treatment of ALI induced by highly pathogenic virus. Herein, we review the literature that links C5a and ALI, and review our understanding of the mechanisms by which C5a affects ALI during highly pathogenic viral infection. In particular, we discuss the potential of the blockade of C5a signaling to treat ALI induced by highly pathogenic viruses.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "20  On 26 May 2015, an imported Middle East respiratory syndrome coronavirus (MERS-CoV) was identified in Guangdong Province, China, and found to be closely related to the MERS-CoV strain prevalent in South Korea. The full genome of the ChinaGD01 strain was sequenced and analyzed to investigate the epidemiology and evolution of MERS-CoV circulating in South Korea and China.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "21  The newly-emerging Middle East respiratory syndrome coronavirus (MERS-CoV) can cause severe and fatal acute respiratory disease in humans. Despite global efforts, the potential for an associated pandemic in the future cannot be excluded. The development of effective counter-measures is urgent. MERS-CoV-specific anti-viral drugs or vaccines are not yet available. Using the spike receptor-binding domain of MERS-CoV (MERS-RBD) to immunize mice, we identified two neutralizing monoclonal antibodies (mAbs) 4C2 and 2E6. Both mAbs potently bind to MERS-RBD and block virus entry in vitro with high efficacy. We further investigated their mechanisms of neutralization by crystallizing the complex between the Fab fragments and the RBD, and solved the structure of the 4C2 Fab/MERS-RBD complex. The structure showed that 4C2 recognizes an epitope that partially overlaps the receptor-binding footprint in MERS-RBD, thereby interfering with the virus/receptor interactions by both steric hindrance and interface-residue competition. 2E6 also blocks receptor binding, and competes with 4C2 for binding to MERS-RBD. Based on the structure, we further humanized 4C2 by preserving only the paratope residues and substituting the remaining amino acids with the counterparts from human immunoglobulins. The humanized 4C2 (4C2h) antibody sustained similar neutralizing activity and biochemical characteristics to the parental mouse antibody. Finally, we showed that 4C2h can significantly abate the virus titers in lungs of Ad5-hCD26-transduced mice infected with MERS-CoV, therefore representing a promising agent for prophylaxis and therapy in clinical settings.                                                                                                                                                                                                                                                             \n",
       "22  Middle East respiratory syndrome coronavirus (MERS-CoV) with pandemic potential is a major worldwide threat to public health. However, vaccine development for this pathogen lags behind as immunity associated with protection is currently largely unknown. In this study, an immunoinformatics-driven genome-wide screening strategy of vaccine targets was performed to thoroughly screen the vital and effective dominant immunogens against MERS-CoV. Conservancy and population coverage analysis of the epitopes were done by the Immune Epitope Database. The results showed that the nucleocapsid (N) protein of MERS-CoV might be a better protective immunogen with high conservancy and potential eliciting both neutralizing antibodies and T-cell responses compared with spike (S) protein. Further, the B-cell, helper T-cell and cytotoxic T lymphocyte (CTL) epitopes were screened and mapped to the N protein. A total of 15 linear and 10 conformal B-cell epitopes that may induce protective neutralizing antibodies were obtained. Additionally, a total of 71 peptides with 9-mer core sequence were identified as helper T-cell epitopes, and 34 peptides were identified as CTL epitopes. Based on the maximum HLA binding alleles, top 10 helper T-cell epitopes and CTL epitopes that may elicit protective cellular immune responses against MERS-CoV were selected as MERS vaccine candidates. Population coverage analysis showed that the putative helper T-cell epitopes and CTL epitopes could cover the vast majority of the population in 15 geographic regions considered where vaccine would be employed. The B- and T-cell stimulation potentials of the screened epitopes is to be further validated for their efficient use as vaccines against MERS-CoV. Collectively, this study provides novel vaccine target candidates and may prompt further development of vaccines against MERS-CoV and other emerging infectious diseases.   \n",
       "\n",
       "   publish_time  \\\n",
       "8   2011-10-23    \n",
       "18  2015-05-06    \n",
       "20  2015-08-13    \n",
       "21  2015-09-22    \n",
       "22  2015-12-07    \n",
       "\n",
       "                                                                                                                                                                                                         authors  \\\n",
       "8   Pang, Xinghuo; Yang, Peng; Li, Shuang; Zhang, Li; Tian, Lili; Li, Yang; Liu, Bo; Zhang, Yi; Liu, Baiwei; Huang, Ruogang; Li, Xinyu; Wang, Quanyi                                                               \n",
       "18  Wang, Renxi; Xiao, He; Guo, Renfeng; Li, Yan; Shen, Beifen                                                                                                                                                     \n",
       "20  Lu, Roujian; Wang, Yanqun; Wang, Wenling; Nie, Kai; Zhao, Yanjie; Su, Juan; Deng, Yao; Zhou, Weimin; Li, Yang; Wang, Huijuan; Wang, Wen; Ke, Changwen; Ma, Xuejun; Wu, Guizhen; Tan, Wenjie                    \n",
       "21  Li, Yan; Wan, Yuhua; Liu, Peipei; Zhao, Jincun; Lu, Guangwen; Qi, Jianxun; Wang, Qihui; Lu, Xuancheng; Wu, Ying; Liu, Wenjun; Zhang, Buchang; Yuen, Kwok-Yung; Perlman, Stanley; Gao, George F; Yan, Jinghua   \n",
       "22  Shi, Jiandong; Zhang, Jing; Li, Sijin; Sun, Jing; Teng, Yumei; Wu, Meini; Li, Jianfan; Li, Yanhan; Hu, Ningzhu; Wang, Haixuan; Hu, Yunzhang                                                                    \n",
       "\n",
       "    has_pdf_parse  has_pmc_xml_parse      full_text_file  \n",
       "8   True           True               custom_license      \n",
       "18  True           True               noncomm_use_subset  \n",
       "20  True           True               comm_use_subset     \n",
       "21  True           True               custom_license      \n",
       "22  True           True               comm_use_subset     "
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_original_metadata= d[df.apply(sum, axis=1).apply(bool)]\n",
    "print(df_original_metadata.shape)\n",
    "df_original_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(52, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>has_pdf_parse</th>\n",
       "      <th>has_pmc_xml_parse</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>asymptomatic</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>immunity</th>\n",
       "      <th>incubation</th>\n",
       "      <th>covid-19</th>\n",
       "      <th>covid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>gl9cjmno</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China</td>\n",
       "      <td>We estimated the attack rate of pandemic (H1N1) 2009 and assessed risk factors for infection among close contacts quarantined in Beijing, People’s Republic of China. The first 613 confirmed cases detected between May 16 and September 15, 2009, were investigated; 7,099 close contacts were located and quarantined. The attack rate of confirmed infection in close contacts was 2.4% overall, ranging from 0.9% among aircraft passengers to &gt;5% among household members. Risk factors for infection among close contacts were younger age, being a household member of an index case-patient, exposure during the index case-patient’s symptomatic phase, and longer exposure. Among close contacts with positive test results at the start of quarantine, 17.2% had subclinical infection. Having contact with a household member and younger age were the major risk factors for acquiring pandemic (H1N1) 2009 influenza virus infection. One person in 6 with confirmed pandemic (H1N1) 2009 was asymptomatic.</td>\n",
       "      <td>2011-10-23</td>\n",
       "      <td>Pang, Xinghuo; Yang, Peng; Li, Shuang; Zhang, Li; Tian, Lili; Li, Yang; Liu, Bo; Zhang, Yi; Liu, Baiwei; Huang, Ruogang; Li, Xinyu; Wang, Quanyi</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>p350i54m</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The role of C5a in acute lung injury induced by highly pathogenic viral infections</td>\n",
       "      <td>The complement system, an important part of innate immunity, plays a critical role in pathogen clearance. Unregulated complement activation is likely to play a crucial role in the pathogenesis of acute lung injury (ALI) induced by highly pathogenic virus including influenza A viruses H5N1, H7N9, and severe acute respiratory syndrome (SARS) coronavirus. In highly pathogenic virus-induced acute lung diseases, high levels of chemotactic and anaphylatoxic C5a were produced as a result of excessive complement activaiton. Overproduced C5a displays powerful biological activities in activation of phagocytic cells, generation of oxidants, and inflammatory sequelae named “cytokine storm”, and so on. Blockade of C5a signaling have been implicated in the treatment of ALI induced by highly pathogenic virus. Herein, we review the literature that links C5a and ALI, and review our understanding of the mechanisms by which C5a affects ALI during highly pathogenic viral infection. In particular, we discuss the potential of the blockade of C5a signaling to treat ALI induced by highly pathogenic viruses.</td>\n",
       "      <td>2015-05-06</td>\n",
       "      <td>Wang, Renxi; Xiao, He; Guo, Renfeng; Li, Yan; Shen, Beifen</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>noncomm_use_subset</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>86t9chxe</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China</td>\n",
       "      <td>On 26 May 2015, an imported Middle East respiratory syndrome coronavirus (MERS-CoV) was identified in Guangdong Province, China, and found to be closely related to the MERS-CoV strain prevalent in South Korea. The full genome of the ChinaGD01 strain was sequenced and analyzed to investigate the epidemiology and evolution of MERS-CoV circulating in South Korea and China.</td>\n",
       "      <td>2015-08-13</td>\n",
       "      <td>Lu, Roujian; Wang, Yanqun; Wang, Wenling; Nie, Kai; Zhao, Yanjie; Su, Juan; Deng, Yao; Zhou, Weimin; Li, Yang; Wang, Huijuan; Wang, Wen; Ke, Changwen; Ma, Xuejun; Wu, Guizhen; Tan, Wenjie</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>kzblr469</td>\n",
       "      <td>PMC</td>\n",
       "      <td>A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein</td>\n",
       "      <td>The newly-emerging Middle East respiratory syndrome coronavirus (MERS-CoV) can cause severe and fatal acute respiratory disease in humans. Despite global efforts, the potential for an associated pandemic in the future cannot be excluded. The development of effective counter-measures is urgent. MERS-CoV-specific anti-viral drugs or vaccines are not yet available. Using the spike receptor-binding domain of MERS-CoV (MERS-RBD) to immunize mice, we identified two neutralizing monoclonal antibodies (mAbs) 4C2 and 2E6. Both mAbs potently bind to MERS-RBD and block virus entry in vitro with high efficacy. We further investigated their mechanisms of neutralization by crystallizing the complex between the Fab fragments and the RBD, and solved the structure of the 4C2 Fab/MERS-RBD complex. The structure showed that 4C2 recognizes an epitope that partially overlaps the receptor-binding footprint in MERS-RBD, thereby interfering with the virus/receptor interactions by both steric hindrance and interface-residue competition. 2E6 also blocks receptor binding, and competes with 4C2 for binding to MERS-RBD. Based on the structure, we further humanized 4C2 by preserving only the paratope residues and substituting the remaining amino acids with the counterparts from human immunoglobulins. The humanized 4C2 (4C2h) antibody sustained similar neutralizing activity and biochemical characteristics to the parental mouse antibody. Finally, we showed that 4C2h can significantly abate the virus titers in lungs of Ad5-hCD26-transduced mice infected with MERS-CoV, therefore representing a promising agent for prophylaxis and therapy in clinical settings.</td>\n",
       "      <td>2015-09-22</td>\n",
       "      <td>Li, Yan; Wan, Yuhua; Liu, Peipei; Zhao, Jincun; Lu, Guangwen; Qi, Jianxun; Wang, Qihui; Lu, Xuancheng; Wu, Ying; Liu, Wenjun; Zhang, Buchang; Yuen, Kwok-Yung; Perlman, Stanley; Gao, George F; Yan, Jinghua</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>gmoq1i5x</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (MERS-CoV) with pandemic potential is a major worldwide threat to public health. However, vaccine development for this pathogen lags behind as immunity associated with protection is currently largely unknown. In this study, an immunoinformatics-driven genome-wide screening strategy of vaccine targets was performed to thoroughly screen the vital and effective dominant immunogens against MERS-CoV. Conservancy and population coverage analysis of the epitopes were done by the Immune Epitope Database. The results showed that the nucleocapsid (N) protein of MERS-CoV might be a better protective immunogen with high conservancy and potential eliciting both neutralizing antibodies and T-cell responses compared with spike (S) protein. Further, the B-cell, helper T-cell and cytotoxic T lymphocyte (CTL) epitopes were screened and mapped to the N protein. A total of 15 linear and 10 conformal B-cell epitopes that may induce protective neutralizing antibodies were obtained. Additionally, a total of 71 peptides with 9-mer core sequence were identified as helper T-cell epitopes, and 34 peptides were identified as CTL epitopes. Based on the maximum HLA binding alleles, top 10 helper T-cell epitopes and CTL epitopes that may elicit protective cellular immune responses against MERS-CoV were selected as MERS vaccine candidates. Population coverage analysis showed that the putative helper T-cell epitopes and CTL epitopes could cover the vast majority of the population in 15 geographic regions considered where vaccine would be employed. The B- and T-cell stimulation potentials of the screened epitopes is to be further validated for their efficient use as vaccines against MERS-CoV. Collectively, this study provides novel vaccine target candidates and may prompt further development of vaccines against MERS-CoV and other emerging infectious diseases.</td>\n",
       "      <td>2015-12-07</td>\n",
       "      <td>Shi, Jiandong; Zhang, Jing; Li, Sijin; Sun, Jing; Teng, Yumei; Wu, Meini; Li, Jianfan; Li, Yanhan; Hu, Ningzhu; Wang, Haixuan; Hu, Yunzhang</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>comm_use_subset</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    cord_uid source_x  \\\n",
       "8   gl9cjmno  PMC       \n",
       "18  p350i54m  PMC       \n",
       "20  86t9chxe  PMC       \n",
       "21  kzblr469  PMC       \n",
       "22  gmoq1i5x  PMC       \n",
       "\n",
       "                                                                                                                                       title  \\\n",
       "8   Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China                                                 \n",
       "18  The role of C5a in acute lung injury induced by highly pathogenic viral infections                                                         \n",
       "20  Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China         \n",
       "21  A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein                              \n",
       "22  Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          abstract  \\\n",
       "8   We estimated the attack rate of pandemic (H1N1) 2009 and assessed risk factors for infection among close contacts quarantined in Beijing, People’s Republic of China. The first 613 confirmed cases detected between May 16 and September 15, 2009, were investigated; 7,099 close contacts were located and quarantined. The attack rate of confirmed infection in close contacts was 2.4% overall, ranging from 0.9% among aircraft passengers to >5% among household members. Risk factors for infection among close contacts were younger age, being a household member of an index case-patient, exposure during the index case-patient’s symptomatic phase, and longer exposure. Among close contacts with positive test results at the start of quarantine, 17.2% had subclinical infection. Having contact with a household member and younger age were the major risk factors for acquiring pandemic (H1N1) 2009 influenza virus infection. One person in 6 with confirmed pandemic (H1N1) 2009 was asymptomatic.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "18  The complement system, an important part of innate immunity, plays a critical role in pathogen clearance. Unregulated complement activation is likely to play a crucial role in the pathogenesis of acute lung injury (ALI) induced by highly pathogenic virus including influenza A viruses H5N1, H7N9, and severe acute respiratory syndrome (SARS) coronavirus. In highly pathogenic virus-induced acute lung diseases, high levels of chemotactic and anaphylatoxic C5a were produced as a result of excessive complement activaiton. Overproduced C5a displays powerful biological activities in activation of phagocytic cells, generation of oxidants, and inflammatory sequelae named “cytokine storm”, and so on. Blockade of C5a signaling have been implicated in the treatment of ALI induced by highly pathogenic virus. Herein, we review the literature that links C5a and ALI, and review our understanding of the mechanisms by which C5a affects ALI during highly pathogenic viral infection. In particular, we discuss the potential of the blockade of C5a signaling to treat ALI induced by highly pathogenic viruses.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "20  On 26 May 2015, an imported Middle East respiratory syndrome coronavirus (MERS-CoV) was identified in Guangdong Province, China, and found to be closely related to the MERS-CoV strain prevalent in South Korea. The full genome of the ChinaGD01 strain was sequenced and analyzed to investigate the epidemiology and evolution of MERS-CoV circulating in South Korea and China.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "21  The newly-emerging Middle East respiratory syndrome coronavirus (MERS-CoV) can cause severe and fatal acute respiratory disease in humans. Despite global efforts, the potential for an associated pandemic in the future cannot be excluded. The development of effective counter-measures is urgent. MERS-CoV-specific anti-viral drugs or vaccines are not yet available. Using the spike receptor-binding domain of MERS-CoV (MERS-RBD) to immunize mice, we identified two neutralizing monoclonal antibodies (mAbs) 4C2 and 2E6. Both mAbs potently bind to MERS-RBD and block virus entry in vitro with high efficacy. We further investigated their mechanisms of neutralization by crystallizing the complex between the Fab fragments and the RBD, and solved the structure of the 4C2 Fab/MERS-RBD complex. The structure showed that 4C2 recognizes an epitope that partially overlaps the receptor-binding footprint in MERS-RBD, thereby interfering with the virus/receptor interactions by both steric hindrance and interface-residue competition. 2E6 also blocks receptor binding, and competes with 4C2 for binding to MERS-RBD. Based on the structure, we further humanized 4C2 by preserving only the paratope residues and substituting the remaining amino acids with the counterparts from human immunoglobulins. The humanized 4C2 (4C2h) antibody sustained similar neutralizing activity and biochemical characteristics to the parental mouse antibody. Finally, we showed that 4C2h can significantly abate the virus titers in lungs of Ad5-hCD26-transduced mice infected with MERS-CoV, therefore representing a promising agent for prophylaxis and therapy in clinical settings.                                                                                                                                                                                                                                                             \n",
       "22  Middle East respiratory syndrome coronavirus (MERS-CoV) with pandemic potential is a major worldwide threat to public health. However, vaccine development for this pathogen lags behind as immunity associated with protection is currently largely unknown. In this study, an immunoinformatics-driven genome-wide screening strategy of vaccine targets was performed to thoroughly screen the vital and effective dominant immunogens against MERS-CoV. Conservancy and population coverage analysis of the epitopes were done by the Immune Epitope Database. The results showed that the nucleocapsid (N) protein of MERS-CoV might be a better protective immunogen with high conservancy and potential eliciting both neutralizing antibodies and T-cell responses compared with spike (S) protein. Further, the B-cell, helper T-cell and cytotoxic T lymphocyte (CTL) epitopes were screened and mapped to the N protein. A total of 15 linear and 10 conformal B-cell epitopes that may induce protective neutralizing antibodies were obtained. Additionally, a total of 71 peptides with 9-mer core sequence were identified as helper T-cell epitopes, and 34 peptides were identified as CTL epitopes. Based on the maximum HLA binding alleles, top 10 helper T-cell epitopes and CTL epitopes that may elicit protective cellular immune responses against MERS-CoV were selected as MERS vaccine candidates. Population coverage analysis showed that the putative helper T-cell epitopes and CTL epitopes could cover the vast majority of the population in 15 geographic regions considered where vaccine would be employed. The B- and T-cell stimulation potentials of the screened epitopes is to be further validated for their efficient use as vaccines against MERS-CoV. Collectively, this study provides novel vaccine target candidates and may prompt further development of vaccines against MERS-CoV and other emerging infectious diseases.   \n",
       "\n",
       "   publish_time  \\\n",
       "8   2011-10-23    \n",
       "18  2015-05-06    \n",
       "20  2015-08-13    \n",
       "21  2015-09-22    \n",
       "22  2015-12-07    \n",
       "\n",
       "                                                                                                                                                                                                         authors  \\\n",
       "8   Pang, Xinghuo; Yang, Peng; Li, Shuang; Zhang, Li; Tian, Lili; Li, Yang; Liu, Bo; Zhang, Yi; Liu, Baiwei; Huang, Ruogang; Li, Xinyu; Wang, Quanyi                                                               \n",
       "18  Wang, Renxi; Xiao, He; Guo, Renfeng; Li, Yan; Shen, Beifen                                                                                                                                                     \n",
       "20  Lu, Roujian; Wang, Yanqun; Wang, Wenling; Nie, Kai; Zhao, Yanjie; Su, Juan; Deng, Yao; Zhou, Weimin; Li, Yang; Wang, Huijuan; Wang, Wen; Ke, Changwen; Ma, Xuejun; Wu, Guizhen; Tan, Wenjie                    \n",
       "21  Li, Yan; Wan, Yuhua; Liu, Peipei; Zhao, Jincun; Lu, Guangwen; Qi, Jianxun; Wang, Qihui; Lu, Xuancheng; Wu, Ying; Liu, Wenjun; Zhang, Buchang; Yuen, Kwok-Yung; Perlman, Stanley; Gao, George F; Yan, Jinghua   \n",
       "22  Shi, Jiandong; Zhang, Jing; Li, Sijin; Sun, Jing; Teng, Yumei; Wu, Meini; Li, Jianfan; Li, Yanhan; Hu, Ningzhu; Wang, Haixuan; Hu, Yunzhang                                                                    \n",
       "\n",
       "    has_pdf_parse  has_pmc_xml_parse      full_text_file  asymptomatic  \\\n",
       "8   True           True               custom_license      1              \n",
       "18  True           True               noncomm_use_subset  0              \n",
       "20  True           True               comm_use_subset     0              \n",
       "21  True           True               custom_license      0              \n",
       "22  True           True               comm_use_subset     0              \n",
       "\n",
       "    coronavirus  immunity  incubation  covid-19  covid  \n",
       "8   0            0         0           0         0      \n",
       "18  1            1         0           0         0      \n",
       "20  1            0         0           0         0      \n",
       "21  1            0         0           0         0      \n",
       "22  1            1         0           0         0      "
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge = pd.concat([df_original_metadata, df_term_search], axis=1)\n",
    "\n",
    "print(df_merge.shape)\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>asymptomatic</th>\n",
       "      <th>coronavirus</th>\n",
       "      <th>immunity</th>\n",
       "      <th>incubation</th>\n",
       "      <th>covid-19</th>\n",
       "      <th>covid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China</td>\n",
       "      <td>2011-10-23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>The role of C5a in acute lung injury induced by highly pathogenic viral infections</td>\n",
       "      <td>2015-05-06</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China</td>\n",
       "      <td>2015-08-13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein</td>\n",
       "      <td>2015-09-22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases</td>\n",
       "      <td>2015-12-07</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Multiple Immunosuppressive Effects of CpG-c41 on Intracellular TLR-Mediated Inflammation</td>\n",
       "      <td>2017-04-30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Altered Lipid Metabolism in Recovered SARS Patients Twelve Years after Infection</td>\n",
       "      <td>2017-08-22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Identification of diverse viruses in upper respiratory samples in dromedary camels from United Arab Emirates</td>\n",
       "      <td>2017-09-13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Diversity of Middle East respiratory syndrome coronaviruses in 109 dromedary camels based on full-genome sequencing, Abu Dhabi, United Arab Emirates</td>\n",
       "      <td>2017-11-08</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Infectious MERS-CoV Isolated From a Mildly Ill Patient, Saudi Arabia</td>\n",
       "      <td>2018-05-15</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Heterosubtypic Protections against Human-Infecting Avian Influenza Viruses Correlate to Biased Cross-T-Cell Responses</td>\n",
       "      <td>2018-08-07</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Angiotensin-converting enzyme 2 (ACE2) proteins of different bat species confer variable susceptibility to SARS-CoV entry</td>\n",
       "      <td>2010-06-22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Short hairpin RNAs targeting M and N genes reduce replication of porcine deltacoronavirus in ST cells</td>\n",
       "      <td>2019-08-28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Isolation of Virus from a SARS Patient and Genome-wide Analysis of Genetic Mutations Related to Pathogenesis and Epidemiology from 47 SARS-CoV Isolates</td>\n",
       "      <td>2005-01-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Molecular basis of binding between novel human coronavirus MERS-CoV and its receptor CD26</td>\n",
       "      <td>2013-07-07</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Scope and extent of healthcare-associated Middle East respiratory syndrome coronavirus transmission during two contemporaneous outbreaks in Riyadh, Saudi Arabia, 2017</td>\n",
       "      <td>2018-12-31</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Severe Acute Respiratory Syndrome-Related Coronavirus Is Inhibited by Interferon-α</td>\n",
       "      <td>2004-04-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Neurotropic virus tracing suggests a membranous‐coating‐mediated mechanism for transsynaptic communication</td>\n",
       "      <td>2013-01-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Dynamics of transmissible gastroenteritis virus internalization unraveled by single‐virus tracking in live cells</td>\n",
       "      <td>2020-02-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Receptor-binding domain of SARS-Cov spike protein: Soluble expression in E.coli, purification and functional characterization</td>\n",
       "      <td>2005-01-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Immunization with Modified Vaccinia Virus Ankara-Based Recombinant Vaccine against Severe Acute Respiratory Syndrome Is Associated with Enhanced Hepatitis in Ferrets</td>\n",
       "      <td>2004-10-26</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Structure of a Conserved Golgi Complex-targeting Signal in Coronavirus Envelope Proteins</td>\n",
       "      <td>2014-03-25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Mutations in a Highly Conserved Motif of nsp1β Protein Attenuate the Innate Immune Suppression Function of Porcine Reproductive and Respiratory Syndrome Virus</td>\n",
       "      <td>2016-01-20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Median infectious dose of human norovirus GII.4 in gnotobiotic pigs is decreased by simvastatin treatment and increased by age</td>\n",
       "      <td>2013-09-01</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Channel-Inactivating Mutations and Their Revertant Mutants in the Envelope Protein of Infectious Bronchitis Virus</td>\n",
       "      <td>2016-12-14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Retroviruses Pseudotyped with the Severe Acute Respiratory Syndrome Coronavirus Spike Protein Efficiently Infect Cells Expressing Angiotensin-Converting Enzyme 2</td>\n",
       "      <td>2004-09-14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Human Coronavirus NL-63 Infections in Children: a 1-Year Study</td>\n",
       "      <td>2005-09-01</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Structure of the Fusion Core and Inhibition of Fusion by a Heptad Repeat Peptide Derived from the S Protein of Middle East Respiratory Syndrome Coronavirus</td>\n",
       "      <td>2013-09-25</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Isolation and growth characterization of novel full length and deletion mutant human MERS-CoV strains from clinical specimens collected during 2015</td>\n",
       "      <td>2019-11-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>A complete sequence and comparative analysis of a SARS-associated virus (Isolate BJ01)</td>\n",
       "      <td>2003</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>Indirect Enzyme-Linked Immunosorbent Assay based on the nucleocapsid protein of SARS-like coronaviruses</td>\n",
       "      <td>2009-04-14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>Promising methods for detection of novel coronavirus SARS‐CoV‐2</td>\n",
       "      <td>2020-03-23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>Coronavirus Disease 2019 (COVID-19): Role of Chest CT in Diagnosis and Management</td>\n",
       "      <td>2020-03-04</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>Severe Acute Respiratory Syndrome Coronavirus 2 from Patient with 2019 Novel Coronavirus Disease, United States</td>\n",
       "      <td>2020-06-01</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>Interpretation of diagnostic laboratory tests for severe acute respiratory syndrome: the Toronto experience</td>\n",
       "      <td>2004-01-06</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>MERS coronavirus envelope protein has a single transmembrane domain that forms pentameric ion channels</td>\n",
       "      <td>2015-04-02</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>Detection of antibodies against SARS-CoV in serum from SARS-infected donors with ELISA and Western blot</td>\n",
       "      <td>2004-11-30</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>Complete Genome Sequences of the SARS-CoV: the BJ Group (Isolates BJ01-BJ04)</td>\n",
       "      <td>2016-11-28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>Evolution and Variation of the SARS-CoV Genome</td>\n",
       "      <td>2016-11-28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>Extraordinary GU-rich single-strand RNA identified from SARS coronavirus contributes an excessive innate immune response</td>\n",
       "      <td>2013-02-28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>Structural Definition of a Unique Neutralization Epitope on the Receptor-Binding Domain of MERS-CoV Spike Glycoprotein</td>\n",
       "      <td>2018-07-10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>Activation of AP-1 signal transduction pathway by SARS coronavirus nucleocapsid protein</td>\n",
       "      <td>2003-11-28</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>Th2 predominance and CD8+ memory T cell depletion in patients with severe acute respiratory syndrome</td>\n",
       "      <td>2005-03-31</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>Diagnosis and Management of First Case of COVID-19 in Canada: Lessons applied from SARS</td>\n",
       "      <td>2020-03-09</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>Coronavirus infection of rat dorsal root ganglia: Ultrastructural characterization of viral replication, transfer, and the early response of satellite cells</td>\n",
       "      <td>2012-02-29</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>Molecular Characteristics, Functions, and Related Pathogenicity of MERS-CoV Proteins</td>\n",
       "      <td>2019-10-31</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>Bat Origins of MERS-CoV Supported by Bat Coronavirus HKU4 Usage of Human Receptor CD26</td>\n",
       "      <td>2014-09-10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>Structural basis for the recognition of SARS-CoV-2 by full-length human ACE2</td>\n",
       "      <td>2020-03-27</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>Analysis of multimerization of the SARS coronavirus nucleocapsid protein</td>\n",
       "      <td>2004-04-02</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>Characterization of protein–protein interactions between the nucleocapsid protein and membrane protein of the SARS coronavirus</td>\n",
       "      <td>2004-10-31</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>Production of a monoclonal antibody against SARS-CoV spike protein with single intrasplenic immunization of plasmid DNA</td>\n",
       "      <td>2005-09-15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>Genome Organization of the SARS-CoV</td>\n",
       "      <td>2016-11-28</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                      title  \\\n",
       "8    Pandemic (H1N1) 2009 among Quarantined Close Contacts, Beijing, People’s Republic of China                                                                               \n",
       "18   The role of C5a in acute lung injury induced by highly pathogenic viral infections                                                                                       \n",
       "20   Complete Genome Sequence of Middle East Respiratory Syndrome Coronavirus (MERS-CoV) from the First Imported MERS-CoV Case in China                                       \n",
       "21   A humanized neutralizing antibody against MERS-CoV targeting the receptor-binding domain of the spike protein                                                            \n",
       "22   Epitope-Based Vaccine Target Screening against Highly Pathogenic MERS-CoV: An In Silico Approach Applied to Emerging Infectious Diseases                                 \n",
       "32   Multiple Immunosuppressive Effects of CpG-c41 on Intracellular TLR-Mediated Inflammation                                                                                 \n",
       "35   Altered Lipid Metabolism in Recovered SARS Patients Twelve Years after Infection                                                                                         \n",
       "38   Identification of diverse viruses in upper respiratory samples in dromedary camels from United Arab Emirates                                                             \n",
       "41   Diversity of Middle East respiratory syndrome coronaviruses in 109 dromedary camels based on full-genome sequencing, Abu Dhabi, United Arab Emirates                     \n",
       "46   Infectious MERS-CoV Isolated From a Mildly Ill Patient, Saudi Arabia                                                                                                     \n",
       "48   Heterosubtypic Protections against Human-Infecting Avian Influenza Viruses Correlate to Biased Cross-T-Cell Responses                                                    \n",
       "64   Angiotensin-converting enzyme 2 (ACE2) proteins of different bat species confer variable susceptibility to SARS-CoV entry                                                \n",
       "65   Short hairpin RNAs targeting M and N genes reduce replication of porcine deltacoronavirus in ST cells                                                                    \n",
       "66   Isolation of Virus from a SARS Patient and Genome-wide Analysis of Genetic Mutations Related to Pathogenesis and Epidemiology from 47 SARS-CoV Isolates                  \n",
       "69   Molecular basis of binding between novel human coronavirus MERS-CoV and its receptor CD26                                                                                \n",
       "71   Scope and extent of healthcare-associated Middle East respiratory syndrome coronavirus transmission during two contemporaneous outbreaks in Riyadh, Saudi Arabia, 2017   \n",
       "72   Severe Acute Respiratory Syndrome-Related Coronavirus Is Inhibited by Interferon-α                                                                                       \n",
       "76   Neurotropic virus tracing suggests a membranous‐coating‐mediated mechanism for transsynaptic communication                                                               \n",
       "77   Dynamics of transmissible gastroenteritis virus internalization unraveled by single‐virus tracking in live cells                                                         \n",
       "87   Receptor-binding domain of SARS-Cov spike protein: Soluble expression in E.coli, purification and functional characterization                                            \n",
       "88   Immunization with Modified Vaccinia Virus Ankara-Based Recombinant Vaccine against Severe Acute Respiratory Syndrome Is Associated with Enhanced Hepatitis in Ferrets    \n",
       "90   Structure of a Conserved Golgi Complex-targeting Signal in Coronavirus Envelope Proteins                                                                                 \n",
       "96   Mutations in a Highly Conserved Motif of nsp1β Protein Attenuate the Innate Immune Suppression Function of Porcine Reproductive and Respiratory Syndrome Virus           \n",
       "98   Median infectious dose of human norovirus GII.4 in gnotobiotic pigs is decreased by simvastatin treatment and increased by age                                           \n",
       "101  Channel-Inactivating Mutations and Their Revertant Mutants in the Envelope Protein of Infectious Bronchitis Virus                                                        \n",
       "102  Retroviruses Pseudotyped with the Severe Acute Respiratory Syndrome Coronavirus Spike Protein Efficiently Infect Cells Expressing Angiotensin-Converting Enzyme 2        \n",
       "106  Human Coronavirus NL-63 Infections in Children: a 1-Year Study                                                                                                           \n",
       "108  Structure of the Fusion Core and Inhibition of Fusion by a Heptad Repeat Peptide Derived from the S Protein of Middle East Respiratory Syndrome Coronavirus              \n",
       "112  Isolation and growth characterization of novel full length and deletion mutant human MERS-CoV strains from clinical specimens collected during 2015                      \n",
       "117  A complete sequence and comparative analysis of a SARS-associated virus (Isolate BJ01)                                                                                   \n",
       "118  Indirect Enzyme-Linked Immunosorbent Assay based on the nucleocapsid protein of SARS-like coronaviruses                                                                  \n",
       "121  Promising methods for detection of novel coronavirus SARS‐CoV‐2                                                                                                          \n",
       "123  Coronavirus Disease 2019 (COVID-19): Role of Chest CT in Diagnosis and Management                                                                                        \n",
       "124  Severe Acute Respiratory Syndrome Coronavirus 2 from Patient with 2019 Novel Coronavirus Disease, United States                                                          \n",
       "130  Interpretation of diagnostic laboratory tests for severe acute respiratory syndrome: the Toronto experience                                                              \n",
       "133  MERS coronavirus envelope protein has a single transmembrane domain that forms pentameric ion channels                                                                   \n",
       "135  Detection of antibodies against SARS-CoV in serum from SARS-infected donors with ELISA and Western blot                                                                  \n",
       "144  Complete Genome Sequences of the SARS-CoV: the BJ Group (Isolates BJ01-BJ04)                                                                                             \n",
       "145  Evolution and Variation of the SARS-CoV Genome                                                                                                                           \n",
       "147  Extraordinary GU-rich single-strand RNA identified from SARS coronavirus contributes an excessive innate immune response                                                 \n",
       "150  Structural Definition of a Unique Neutralization Epitope on the Receptor-Binding Domain of MERS-CoV Spike Glycoprotein                                                   \n",
       "154  Activation of AP-1 signal transduction pathway by SARS coronavirus nucleocapsid protein                                                                                  \n",
       "156  Th2 predominance and CD8+ memory T cell depletion in patients with severe acute respiratory syndrome                                                                     \n",
       "158  Diagnosis and Management of First Case of COVID-19 in Canada: Lessons applied from SARS                                                                                  \n",
       "161  Coronavirus infection of rat dorsal root ganglia: Ultrastructural characterization of viral replication, transfer, and the early response of satellite cells             \n",
       "162  Molecular Characteristics, Functions, and Related Pathogenicity of MERS-CoV Proteins                                                                                     \n",
       "165  Bat Origins of MERS-CoV Supported by Bat Coronavirus HKU4 Usage of Human Receptor CD26                                                                                   \n",
       "168  Structural basis for the recognition of SARS-CoV-2 by full-length human ACE2                                                                                             \n",
       "170  Analysis of multimerization of the SARS coronavirus nucleocapsid protein                                                                                                 \n",
       "171  Characterization of protein–protein interactions between the nucleocapsid protein and membrane protein of the SARS coronavirus                                           \n",
       "172  Production of a monoclonal antibody against SARS-CoV spike protein with single intrasplenic immunization of plasmid DNA                                                  \n",
       "174  Genome Organization of the SARS-CoV                                                                                                                                      \n",
       "\n",
       "    publish_time  asymptomatic  coronavirus  immunity  incubation  covid-19  \\\n",
       "8    2011-10-23   1             0            0         0           0          \n",
       "18   2015-05-06   0             1            1         0           0          \n",
       "20   2015-08-13   0             1            0         0           0          \n",
       "21   2015-09-22   0             1            0         0           0          \n",
       "22   2015-12-07   0             1            1         0           0          \n",
       "32   2017-04-30   0             0            1         0           0          \n",
       "35   2017-08-22   0             1            0         0           0          \n",
       "38   2017-09-13   0             1            0         0           0          \n",
       "41   2017-11-08   0             1            0         1           0          \n",
       "46   2018-05-15   1             1            0         0           0          \n",
       "48   2018-08-07   0             0            1         0           0          \n",
       "64   2010-06-22   0             1            0         0           0          \n",
       "65   2019-08-28   0             1            0         0           0          \n",
       "66   2005-01-01   0             1            0         0           0          \n",
       "69   2013-07-07   0             2            0         0           0          \n",
       "71   2018-12-31   0             1            0         0           0          \n",
       "72   2004-04-01   0             1            0         0           0          \n",
       "76   2013-01-01   0             1            0         0           0          \n",
       "77   2020-02-04   0             1            0         0           0          \n",
       "87   2005-01-01   0             1            0         0           0          \n",
       "88   2004-10-26   0             1            0         0           0          \n",
       "90   2014-03-25   0             1            0         0           0          \n",
       "96   2016-01-20   0             0            1         0           0          \n",
       "98   2013-09-01   0             0            2         0           0          \n",
       "101  2016-12-14   0             1            0         0           0          \n",
       "102  2004-09-14   0             1            0         0           0          \n",
       "106  2005-09-01   0             2            0         0           0          \n",
       "108  2013-09-25   0             1            0         0           0          \n",
       "112  2019-11-01   0             1            0         0           0          \n",
       "117  2003         0             1            0         0           0          \n",
       "118  2009-04-14   0             1            0         0           0          \n",
       "121  2020-03-23   0             1            0         0           0          \n",
       "123  2020-03-04   0             1            0         0           12         \n",
       "124  2020-06-01   0             1            0         0           0          \n",
       "130  2004-01-06   0             1            0         0           0          \n",
       "133  2015-04-02   0             1            0         0           0          \n",
       "135  2004-11-30   0             1            0         0           0          \n",
       "144  2016-11-28   0             1            0         0           0          \n",
       "145  2016-11-28   0             1            0         0           0          \n",
       "147  2013-02-28   0             0            1         0           0          \n",
       "150  2018-07-10   0             1            0         0           0          \n",
       "154  2003-11-28   0             3            0         0           0          \n",
       "156  2005-03-31   0             1            2         0           0          \n",
       "158  2020-03-09   0             1            0         0           2          \n",
       "161  2012-02-29   0             1            0         0           0          \n",
       "162  2019-10-31   0             1            1         0           0          \n",
       "165  2014-09-10   0             1            0         0           0          \n",
       "168  2020-03-27   0             4            0         0           1          \n",
       "170  2004-04-02   0             4            0         0           0          \n",
       "171  2004-10-31   0             2            0         0           0          \n",
       "172  2005-09-15   0             1            0         0           0          \n",
       "174  2016-11-28   0             2            0         0           0          \n",
       "\n",
       "     covid  \n",
       "8    0      \n",
       "18   0      \n",
       "20   0      \n",
       "21   0      \n",
       "22   0      \n",
       "32   0      \n",
       "35   0      \n",
       "38   0      \n",
       "41   0      \n",
       "46   0      \n",
       "48   0      \n",
       "64   0      \n",
       "65   0      \n",
       "66   0      \n",
       "69   0      \n",
       "71   0      \n",
       "72   0      \n",
       "76   0      \n",
       "77   0      \n",
       "87   0      \n",
       "88   0      \n",
       "90   0      \n",
       "96   0      \n",
       "98   0      \n",
       "101  0      \n",
       "102  0      \n",
       "106  0      \n",
       "108  0      \n",
       "112  0      \n",
       "117  0      \n",
       "118  0      \n",
       "121  3      \n",
       "123  0      \n",
       "124  0      \n",
       "130  0      \n",
       "133  0      \n",
       "135  0      \n",
       "144  0      \n",
       "145  0      \n",
       "147  0      \n",
       "150  0      \n",
       "154  0      \n",
       "156  0      \n",
       "158  0      \n",
       "161  0      \n",
       "162  0      \n",
       "165  0      \n",
       "168  0      \n",
       "170  0      \n",
       "171  0      \n",
       "172  0      \n",
       "174  0      "
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge[[\"title\", \"publish_time\"] + useful_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer\n",
    "from sklearn.decomposition import NMF, LatentDirichletAllocation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_samples = 2000\n",
    "n_features = 1000\n",
    "\n",
    "n_components = 10\n",
    "n_top_words = 20\n",
    "\n",
    "print(\"Fitting LDA models with tf features, n_samples=%d and n_features=%d...\" % (n_samples, n_features))\n",
    "\n",
    "lda = LatentDirichletAllocation(n_components=n_components, \n",
    "                                max_iter=5,\n",
    "                                learning_method='online',\n",
    "                                learning_offset=50.,\n",
    "                                random_state=0)\n",
    "t0 = time()\n",
    "lda.fit(tf)\n",
    "print(\"done in %0.3fs.\" % (time() - t0))\n",
    "\n",
    "print(\"\\nTopics in LDA model:\")\n",
    "tf_feature_names = tf_vectorizer.get_feature_names()\n",
    "print_top_words(lda, tf_feature_names, n_top_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "vectorizer2 = CountVectorizer(analyzer='word', ngram_range=(2, 2))\n",
    "X2 = vectorizer2.fit_transform(corpus)\n",
    "print(vectorizer2.get_feature_names())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10', '109', '13', '14', '145', '15', '18', '19', '1β', '2005', '2006', '2007', '2008', '2009', '2015', '2017', '2019', '47', '63', '795', '90β', 'abbreviation', 'abu', 'ace2', 'acetyl', 'acid', 'acids', 'acquired', 'action', 'activation', 'activities', 'activity', 'acute', 'additional', 'adenovirus', 'adenoviruses', 'adjuvant', 'adjuvanted', 'adult', 'af', 'african', 'after', 'against', 'age', 'alberta', 'allele', 'along', 'alphavirus', 'alteration', 'altered', 'ameliorates', 'america', 'among', 'amplification', 'an', 'analyses', 'analysis', 'anchorless', 'and', 'angiogenic', 'angiotensin', 'ankara', 'antagonist', 'antenna', 'anti', 'antibodies', 'antibody', 'anticancer', 'antigens', 'antimicrobial', 'antiviral', 'antivirals', 'ap', 'apoptosis', 'apoptotic', 'application', 'applications', 'applied', 'approach', 'arab', 'arabia', 'areas', 'array', 'arrest', 'arteritis', 'arterivirus', 'artistic', 'as', 'aspua', 'assay', 'assays', 'assembled', 'assisted', 'associated', 'association', 'astrocytes', 'asymptomatic', 'asynchronously', 'attenuate', 'attenuated', 'attenuates', 'atypical', 'automatic', 'autophagy', 'avian', 'axis', 'bacillus', 'bacteria', 'bacterium', 'band', 'base', 'based', 'basic', 'basis', 'bat', 'bats', 'bcl', 'beijing', 'beta', 'between', 'biased', 'bicelle', 'big', 'binding', 'binds', 'bioactivity', 'biology', 'biomedical', 'biotinylated', 'birds', 'bj', 'bj01', 'bj04', 'bleomycin', 'block', 'blocking', 'blot', 'bocavirus', 'botanical', 'both', 'bound', 'broadly', 'bronchitis', 'building', 'bx', 'by', 'c41', 'c5a', 'camels', 'canada', 'canadian', 'cancer', 'care', 'case', 'cases', 'cathepsin', 'causing', 'cd26', 'cd4', 'cd8', 'cell', 'cells', 'cellular', 'chain', 'channel', 'channels', 'characterisation', 'characteristics', 'characterization', 'chest', 'children', 'china', 'chinese', 'chlorine', 'chondroitin', 'chop', 'chronic', 'circular', 'circulating', 'cities', 'clinical', 'close', 'closed', 'cluster', 'co', 'coating', 'codon', 'coli', 'collected', 'combined', 'communication', 'community', 'comparative', 'complement', 'complete', 'complex', 'compound', 'comprehensive', 'confer', 'conferring', 'confiscations', 'conjugate', 'conservation', 'conserved', 'construction', 'contacts', 'containing', 'contemporaneous', 'contributes', 'control', 'converting', 'coordinated', 'copper', 'core', 'coronavirus', 'coronaviruses', 'correlate', 'cortical', 'cov', 'covid', 'cpg', 'cross', 'crystal', 'ct', 'culture', 'cxcl9', 'cycle', 'cyclodextrin', 'cyclophilin', 'cytokines', 'cytotoxic', 'damage', 'dampened', 'data', 'decontamination', 'decorated', 'decreased', 'defense', 'defensin', 'definition', 'deletion', 'delhi', 'delineating', 'delivery', 'deltacoronavirus', 'dependent', 'depletion', 'deployable', 'derivatives', 'derived', 'design', 'detect', 'detection', 'deubiquitinase', 'development', 'dhabi', 'diagnosis', 'diagnostic', 'diarrhea', 'different', 'dioxide', 'discovery', 'disease', 'diseases', 'disorder', 'dissection', 'distinct', 'distribution', 'disturbance', 'diverse', 'diversity', 'dna', 'does', 'domain', 'dominant', 'donors', 'dorsal', 'dose', 'dot', 'dr', 'dromedary', 'drug', 'drugs', 'dual', 'due', 'duplex', 'during', 'dynamics', 'early', 'east', 'eastern', 'effect', 'effective', 'effects', 'efficient', 'efficiently', 'eif2α', 'eight', 'elderly', 'elicit', 'elisa', 'emergencies', 'emerging', 'emirates', 'enables', 'encephalitis', 'endoplasmic', 'endothelial', 'enhanced', 'enhances', 'enrichment', 'enterovirus', 'entry', 'envelope', 'environment', 'enzyme', 'epidemic', 'epidemiology', 'epitope', 'equine', 'es2', 'escherichia', 'essential', 'evaluation', 'evolution', 'evolutionarily', 'excessive', 'expansion', 'experience', 'expressing', 'expression', 'expressional', 'extent', 'extracellular', 'extraordinary', 'fabrication', 'factors', 'failure', 'father', 'fbps', 'fc', 'fecal', 'ferrets', 'fibronectin', 'fibrosis', 'fidelity', 'field', 'fighter', 'first', 'flood', 'follow', 'following', 'foot', 'for', 'forms', 'formula', 'four', 'frameshifting', 'from', 'full', 'fulminant', 'function', 'functional', 'functions', 'fusion', 'g0', 'g1', 'g14', 'g3', 'ganglia', 'gas', 'gastric', 'gastroenteritis', 'gastrointestinal', 'gc', 'gd', 'gene', 'general', 'generated', 'generation', 'genes', 'genetic', 'genome', 'genomic', 'genotype', 'genotyping', 'gii', 'glucosyltransferase', 'glycoprotein', 'glycyrrhetinic', 'glycyrrhizin', 'gn', 'gnotobiotic', 'golgi', 'gondii', 'gram', 'graphene', 'green', 'group', 'growing', 'growth', 'gu', 'guidance', 'gut', 'guyana', 'h1n1', 'h5n1', 'h7n9', 'ha', 'hairpin', 'has', 'hbv', 'health', 'healthcare', 'hearing', 'heat', 'hela', 'hemagglutinin', 'hepatic', 'hepatitis', 'hepatocyte', 'heptad', 'herpes', 'heterogeneous', 'heteropolytungstate', 'heterosubtypic', 'high', 'higher', 'highly', 'histoplasma', 'hiv', 'hku4', 'horses', 'hospitalized', 'host', 'hsv', 'hubei', 'human', 'humanized', 'hunan', 'huo', 'hydrogen', 'hydrophobic', 'hyper', 'hypothesis', 'identification', 'identified', 'ii', 'iia', 'ill', 'illnesses', 'immune', 'immunization', 'immunogenicity', 'immunological', 'immunosorbent', 'immunosuppressive', 'implications', 'imported', 'improves', 'in', 'inactivated', 'inactivating', 'increased', 'independent', 'indirect', 'induce', 'induced', 'induces', 'induction', 'infect', 'infected', 'infecting', 'infection', 'infections', 'infectious', 'inflammation', 'influence', 'influenza', 'inhibited', 'inhibiting', 'inhibition', 'inhibitor', 'inhibitors', 'inhibits', 'injury', 'innate', 'inositol', 'ins29', 'insectivorous', 'insulated', 'integrated', 'interactions', 'interactome', 'interfering', 'interferon', 'interleukin', 'internalization', 'interpretation', 'intracellular', 'intrasplenic', 'intrinsic', 'invasion', 'ion', 'ire1', 'ire1α', 'is', 'ischemia', 'isolate', 'isolated', 'isolates', 'isolation', 'isothermal', 'its', 'janus', 'japanese', 'jia', 'jnk', 'key', 'kinase', 'kingdom', 'labeled', 'labeling', 'laboratories', 'laboratory', 'lack', 'lactamase', 'language', 'leader', 'leads', 'length', 'lesions', 'lessons', 'lien', 'like', 'lineage', 'linkage', 'linked', 'lipid', 'lipopolysaccharide', 'live', 'lmpg', 'local', 'localization', 'loop', 'loss', 'lung', 'macaques', 'macrophage', 'main', 'mainland', 'mammalian', 'management', 'managing', 'many', 'map', 'mapk', 'marc', 'matthews', 'mechanism', 'mechanisms', 'median', 'mediated', 'membrane', 'membranous', 'memory', 'mers', 'metabolism', 'metabolite', 'metagenomic', 'metagenomics', 'metallo', 'metapneumovirus', 'metapneumoviruses', 'method', 'methods', 'mf59', 'mice', 'micelles', 'microarray', 'microarrays', 'microbiology', 'microcirculation', 'microglia', 'microtubules', 'middle', 'mildly', 'milk', 'mimivirus', 'mining', 'mitochondrial', 'mode', 'model', 'modified', 'modulating', 'molecular', 'molecule', 'monkeys', 'monoclonal', 'monotreme', 'mosaic', 'motif', 'mouth', 'moving', 'mucosal', 'multi', 'multienzyme', 'multimerization', 'multiple', 'multiplex', 'mutant', 'mutants', 'mutation', 'mutations', 'nanobody', 'nanoparticle', 'nanoparticles', 'nanoshells', 'natural', 'nectin', 'neonatal', 'netncsp', 'neuraminidase', 'neurologic', 'neurons', 'neurotropic', 'neutralising', 'neutralization', 'neutralizing', 'neutrophil', 'new', 'newcastle', 'news', 'next', 'nf', 'niger', 'nipah', 'nl', 'non', 'nonoverlapping', 'nonstructural', 'norovirus', 'north', 'not', 'notoginsenoside', 'novel', 'ns', 'nsp1β', 'nsp2', 'nuclear', 'nucleocapsid', 'of', 'oleanolic', 'oligosaccharides', 'olivaceus', 'on', 'one', 'online', 'opme', 'opposite', 'optimized', 'or', 'organ', 'organization', 'origin', 'origins', 'oseltamivir', 'other', 'outbreaks', 'outcome', 'outcomes', 'over', 'oxide', 'p38', 'pages', 'pallidum', 'pams', 'pandemic', 'panel', 'paralichthys', 'paramyxovirus', 'paramyxoviruses', 'paraquat', 'particle', 'partly', 'patch', 'pathogenesis', 'pathogenic', 'pathogenicity', 'pathotypic', 'pathway', 'pathways', 'patient', 'patients', 'pattern', 'pcr', 'pdk1', 'pentameric', 'people', 'peptide', 'pharmacology', 'phlebovirus', 'phosphorylation', 'phylogenetic', 'pigeon', 'pigeons', 'pigs', 'pilot', 'pipeline', 'plague', 'plasmid', 'plasmonic', 'pneumonia', 'polarization', 'poly', 'polymerase', 'polysaccharides', 'polyvalent', 'porcine', 'positive', 'post', 'pot', 'potential', 'practical', 'predominance', 'prefix', 'pregnancy', 'pregnane', 'preparation', 'pressure', 'prevalence', 'prevention', 'primate', 'production', 'profile', 'profiling', 'programmed', 'progression', 'proliferation', 'promising', 'promote', 'promotes', 'properties', 'prophylaxis', 'prospects', 'prostate', 'protect', 'protections', 'protective', 'protein', 'proteins', 'proteomic', 'proteomics', 'proximal', 'pseudotyped', 'public', 'pulmonary', 'purification', 'pyrosequencing', 'qi', 'qpcr', 'quantitative', 'quantum', 'quarantined', 'r1', 'range', 'rapid', 'rash', 'rat', 'rats', 'reaction', 'reactive', 'real', 'receptor', 'recognition', 'recombinant', 'recovered', 'recovery', 'reduce', 'region', 'regions', 'regulated', 'related', 'renal', 'reovirus', 'repeat', 'reperfusion', 'replication', 'reports', 'reproductive', 'republic', 'requiring', 'resistance', 'resistant', 'respiratory', 'response', 'responses', 'reticulum', 'retroviruses', 'reveal', 'reveals', 'reverse', 'revertant', 'rhinovirus', 'ribosomal', 'ribosomes', 'rich', 'ridge', 'rise', 'risk', 'riyadh', 'rna', 'rnas', 'rnase', 'role', 'roles', 'root', 'ros', 'rotaviruses', 'route', 'routes', 'rt', 'safety', 'samples', 'sargassum', 'sars', 'satellite', 'saudi', 'scale', 'scavenger', 'schiff', 'school', 'scope', 'screening', 'season', 'seco', 'secretion', 'sectional', 'selected', 'self', 'sensitive', 'sentiment', 'sequence', 'sequences', 'sequencing', 'sequential', 'series', 'serum', 'severe', 'shaped', 'shapes', 'shock', 'short', 'show', 'shutoff', 'sialic', 'sialosides', 'signal', 'signaling', 'silico', 'silver', 'simarteriviruses', 'simplex', 'simultaneous', 'simultaneously', 'simvastatin', 'single', 'site', 'skin', 'small', 'social', 'soluble', 'south', 'spatial', 'species', 'specific', 'specimens', 'spike', 'spores', 'st', 'stage', 'states', 'stem', 'step', 'steroids', 'stimulated', 'sting', 'strains', 'strand', 'strategy', 'streptococcus', 'stress', 'structural', 'structure', 'structures', 'study', 'subgenomic', 'subjects', 'substituted', 'substrate', 'subtilis', 'subunit', 'suckling', 'suffering', 'suggests', 'suis', 'sulfate', 'sulfide', 'support', 'supported', 'suppression', 'surfaces', 'surveillance', 'susceptibility', 'swine', 'syncytial', 'syndrome', 'synthesis', 'synthesize', 'system', 'systemic', 'systems', 'tanshinone', 'taqman', 'target', 'targeted', 'targeting', 'targets', 'taxonomic', 'teh', 'temperature', 'temporal', 'tests', 'th2', 'than', 'that', 'the', 'their', 'through', 'time', 'titanium', 'tlr', 'tlr3', 'to', 'tobacco', 'toll', 'toronto', 'toxoplasma', 'tracing', 'tracking', 'tract', 'trade', 'transactivation', 'transcription', 'transduction', 'transfer', 'translation', 'transmembrane', 'transmissible', 'transmission', 'transsynaptic', 'traps', 'traumatic', 'treatment', 'trim22', 'tuberculosis', 'tumor', 'twelve', 'two', 'type', 'ultrastructural', 'uncoating', 'uncovers', 'unexplained', 'unique', 'united', 'universal', 'unraveled', 'untranslated', 'up', 'upper', 'urban', 'usage', 'using', 'usp9x', 'vaccination', 'vaccine', 'vaccines', 'vaccinia', 'var', 'variable', 'variation', 'variations', 'vero', 'via', 'vip', 'viral', 'virions', 'viroporins', 'virulence', 'virus', 'viruses', 'vitro', 'voids', 'web', 'western', 'wide', 'wind', 'with', 'without', 'women', 'workers', 'wu', 'xbp1', 'xiang', 'year', 'years', 'zheng', 'zika', 'κb', 'μ1', 'σ1']\n",
      "(176, 1022)\n"
     ]
    }
   ],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "\n",
    "vectorizer = TfidfVectorizer()\n",
    "X = vectorizer.fit_transform(corpus)\n",
    "print(vectorizer.get_feature_names())\n",
    "print(X.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Topics in NMF model (Frobenius norm):\n",
      "\n",
      "Topic #0: reproductive porcine syndrome virus respiratory and of proteins by replication \n",
      "\n",
      "Topic #1: cov sars genome the of mers complete and organization isolates \n",
      "\n",
      "Topic #2: in with china of respiratory viruses and disease children acute \n",
      "\n",
      "Topic #3: detection for novel pcr sensitive of diarrhea epidemic rapid virus \n",
      "\n",
      "Topic #4: protein nucleocapsid the of coronavirus sars interactions multimerization membrane analysis \n",
      "\n"
     ]
    }
   ],
   "source": [
    "n_components =5\n",
    "n_top_words = 10\n",
    "\n",
    "def print_top_words(model, feature_names, n_top_words):\n",
    "    for topic_idx, topic in enumerate(model.components_):\n",
    "        message = \"Topic #%d: \" % topic_idx\n",
    "        message += \" \".join([feature_names[i] for i in topic.argsort()[:-n_top_words - 1:-1]])\n",
    "        print(message, \"\\n\")\n",
    "\n",
    "\n",
    "from sklearn.decomposition import NMF\n",
    "nmf = NMF(n_components=n_components, random_state=1, alpha=.1, l1_ratio=.5).fit(X)\n",
    "\n",
    "print(\"\\nTopics in NMF model (Frobenius norm):\\n\")\n",
    "tfidf_feature_names = vectorizer.get_feature_names()\n",
    "print_top_words(nmf, tfidf_feature_names, n_top_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
